### PR TITLE
Move all API related annotations into annotation_key_constants.go

### PR DESCRIPTION
**Pull request by MrHohn**
Separate from #45869. See https://github.com/kubernetes/kubernetes/pull/45869#discussion_r116839411 for details.

This PR does nothing but move constants around :)

/assign @caesarxuchao 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by MrHohn** [comment]
@caesarxuchao Sorry, the failure is hack/.linted_packages. The stale entry (pkg/api/annotations) is removed now.

**Comment by MrHohn** [comment]
@k8s-bot kubemark e2e test this

**Comment by MrHohn** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by MrHohn** [comment]
/assign @thockin 
For approval :)

(This is blocking #45869.)

File: pkg/api/annotation_key_constants.go [hunk 1]
```diff
@@ -70,4 +72,35 @@ const (
...
+	LastAppliedConfigAnnotation = kubectlPrefix + "last-applied-configuration"
+
+	// AnnotationLoadBalancerSourceRangesKey is the key of the annotation on a service to set allowed ingress ranges on their LoadBalancers
+	//
+	// It should be a comma-separated list of CIDRs, e.g. `0.0.0.0/0` to
+	// allow full access (the default) or `18.0.0.0/8,56.0.0.0/8` to allow
+	// access only from the CIDRs currently allocated to MIT & the USPS.
+	//
+	// Not all cloud providers support this annotation, though AWS & GCE do.
+	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
```
Comment (thockin) [inline pkg/api/annotation_key_constants.go]: if you're going to extract `kubectlPrefix` why not `serviceBetaPrefix`

**Review by thockin — COMMENTED** [review]
This is better, but I'm still not satisfied :)

File: pkg/api/annotation_key_constants.go (hunk 1 above)
Comment (MrHohn) [inline pkg/api/annotation_key_constants.go]: Thanks! Yeah we could do that later :)

**Pull request merged**